### PR TITLE
Added new operational_flows.rst page

### PR DESCRIPTION
**Pull request by Olga-MSFT**
Added new page for Office for iOS documentation to explain the implementation details behind the user scenarios. Updated the index.rst file to add this new page to the TOC.

### CODE REVIEW

File: docs/native/src/scenarios/operational_flows.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
+..  _operational flows:
+
+Operational flows
+=================
+
+This section outlines the implementation requirements and operational flows for user scenarios outlined in :ref:`Browsing, opening and saving files<browse files>` section. 
+
+**Terminology**
```
Comment (tylerbutler) [inline docs/native/src/scenarios/operational_flows.rst]: Just make this a [definition list](http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html#definition-lists). You shouldn't format this stuff yourself.

File: docs/native/src/scenarios/operational_flows.rst [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ * *Office Client:* The |Office iOS| app, e.g. Excel/Word/PPT 
+ * *Office Identity Service:* The Office Identity service which is Office's service for handling user identity information. In OAuth flows, the Office Identity Service behaves as the External User Agent
+ * *Bootstrapper URL:* URL which the Storage Provider hosts to allow an Office Client app with a valid Oauth 2.0 access token to retrieve a WOPI access token
+ * *Token Endpoint URL:* URL which the Storage Provider hosts & uses in order to generate access or refresh tokens 
+ * *Services Manager:* Office-owned service which stores a Service Catalog, which has all of the available services for an app or endpoint
+ * *Storage provider:* Refers to a CSPP partner who has integrated their service into |Office iOS|
+
+ ..  note::
+	 * Any time during these flows that we try to hit the bootstrapper and have expired or invalid OAuth credentials we will ask the user to log in again.
+	 * Any time during these flows that we have a missing, invalid, or expired WOPI access token we will attempt to get a valid WOPI access_token by calling :ref:`GetNewAccessToken on the Bootstrapper <GetRootContainer (bootstrapper)>`.
```
Comment (tylerbutler) [inline docs/native/src/scenarios/operational_flows.rst]: I think the link here is wrong. I think you want:

> ...we will attempt to get a valid WOPI access token by calling ``:ref:`GetNewAccessToken` ``.

Comment (tylerbutler) [inline docs/native/src/scenarios/operational_flows.rst]: Also, note that when using `:ref:` (or any other similar constructs like `:term`), you don't need to specify a title unless you want it to be different than the title of the section you're referring to. In general you shouldn't specify your own title when referring to a section unless you really need to to make the language sound right.

File: docs/native/src/scenarios/operational_flows.rst [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+ #. The Office Identity Service calls the Bootstrapper using the tokens to retrieve the user profile information. 
+ #. The Office Identity service sends these tokens & the user profile information to the client. 
+ #. The client uses the OAuth code to call the bootstrap shortcut to obtain a Root Container URL. If the shortcut is not implemented, it calls the Ecosystem URL.
+ #. The Bootstrapper returns the root container or ecosystem URL, as well as the OAuth token info and profile information to the Office client.
+ #. The user is now able to see their files under the Storage Provider's Place and open the Storage Provider files directly in their |Office iOS| app. 
+
+
+Browsing and opening files
+--------------------------
+Here is the operational flow for browsing and opening files.
```
Comment (tylerbutler) [inline docs/native/src/scenarios/operational_flows.rst]: It seems unnecessary to describe in this level of detail how Office for iOS uses WOPI. It'll almost certainly become out of date, and really shouldn't be relevant in the vast majority of cases. It makes sense for the auth piece, since we don't yet cover how auth is supposed to work with the Bootstrapper. But this level of detail, IMO, will likely cause even more confusion.

Comment (Olga-MSFT) [inline docs/native/src/scenarios/operational_flows.rst]: We got enough questions on it that we thought it would be useful to document but, if it's too much effort to maintain or winds up hurting more than it helps, I'd be open everything after the auth flow.

File: docs/native/src/scenarios/operational_flows.rst (hunk 1 above)
Comment (Olga-MSFT) [inline docs/native/src/scenarios/operational_flows.rst]: Updated this in the latest commit.

File: docs/native/src/scenarios/operational_flows.rst (hunk 2 above)
Comment (Olga-MSFT) [inline docs/native/src/scenarios/operational_flows.rst]: Updated this in the latest commit.

File: docs/native/src/index.rst [hunk 4]
```diff
@@ -49,13 +49,13 @@ To integrate your application with |Office iOS|, you need to do the following:
...
     /scenarios/browse
     /scenarios/from_app
+    /scenarios/operational_flows
 
 ..  toctree::
     :maxdepth: 2
     :glob:
     :hidden:
     :caption: Onboarding
-    :name: onboardingtoc
```
Comment (tylerbutler) [inline docs/native/src/index.rst]: why did you remove this? You'll likely get a build error because the name conflicts with another name.

File: docs/native/src/scenarios/operational_flows.rst (hunk 3 above)
Comment (tylerbutler) [inline docs/native/src/scenarios/operational_flows.rst]: OK. In that case, I suggest rewording some stuff so that it's clear that this is a description of Office iOS behavior. As it is, it's a little ambiguous, and a quick read will potentially mislead readers into thinking that this is documentation for things they (the host) need to do. It sort of reads like instructions on how to build a WOPI client. For example:

> If the user has made changes to the file, update the file's contents using PutFile for the File URL. The PutFile request must contain the current WOPI Lock Token for the locked file or it will fail.

Is this something I should do as a host? I think just adding a subject to the sentences will clarify it a lot. For example:

> If the user has made changes to the file, Office for iOS will the file's contents by sending a PutFile request to the File URL. The PutFile request will contain the current WOPI Lock ID for the locked file.

File: docs/native/src/scenarios/operational_flows.rst [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+
+Browsing and opening files
+--------------------------
+Here is the operational flow for browsing and opening files.   
+
+#. *Get the RootContainerURL:* 
+
+    * Option 1: Call the :ref:`GetRootContainer (bootstrapper)` shortcut and provide all of the required Auth headers. 
+    * Option 2: If the :ref:`GetRootContainer (bootstrapper)` shortcut is not implemented, call :ref:`GetRootContainer (ecosystem)`. If this call doesn't contain ContainerInfo, use the WOPI URL to call :ref:`CheckContainerInfo`.
```
Comment (tylerbutler) [inline docs/native/src/scenarios/operational_flows.rst]: Elsewhere in the docs, it says that the GetRootContainer shortcut operation is required: http://wopi.readthedocs.io/projects/officewopi/en/latest/wopi_requirements.html#bootstrapper

Comment (karvell) [inline docs/native/src/scenarios/operational_flows.rst]: We should update http://wopi.readthedocs.io/projects/officewopi/en/latest/wopi_requirements.html#bootstrapper to be more clear. For the shortcuts, when it says required, it really just means we make sure if it if it's available, while we don't yet use the GetFileWopiSrc shortcuts even if implemented.

File: docs/native/src/index.rst (hunk 4 above)
Comment (Olga-MSFT) [inline docs/native/src/index.rst]: That was inadvertent-- added it back.

File: docs/native/src/scenarios/operational_flows.rst (hunk 5 above)
Comment (Olga-MSFT) [inline docs/native/src/scenarios/operational_flows.rst]: I updated this in the WOPI requirements page

File: docs/native/src/wopi_requirements.rst [hunk 6]
```diff
@@ -53,7 +53,7 @@ Bootstrapper
 
 * :ref:`Bootstrap`
 * :ref:`GetNewAccessToken`
-* :ref:`Shortcut operations` (Of these, only :ref:`GetRootContainer (bootstrapper)` is required)
+* :ref:`GetRootContainer (bootstrapper)`, if :ref:`GetRootContainer (ecosystem)` isn't implemeneted
```
Comment (tylerbutler) [inline docs/native/src/wopi_requirements.rst]: This still doesn't seem accurate. Why not just list GetRootContainer (bootstrapper) as a firm requirement? I don't think you want partners _not_ implementing it, even though you can technically fall back to GetRootContainer (ecosystem).

Alternatively, add it to the 'not called currently but you still need to implement it' list below.

File: docs/native/src/wopi_requirements.rst [hunk 7]
```diff
@@ -5,7 +5,7 @@ WOPI implementation requirements for |Office iOS| integration
 =============================================================
```
Comment (tylerbutler) [inline docs/native/src/wopi_requirements.rst]: I think my biggest problem with this section as a whole is that it _implies_ hosts don't have to implement operations that Office for iOS doesn't currently call. But that's not accurate. If we're expecting partners to implement everything (and I think we are), then listing out what operations are actually called today doesn't really serve a purpose IMO.

**Comment by tylerbutler** [comment]
For future changes, can you set the line width of your editor to 120 characters and add line breaks so each line is <120? I can clean that up for this PR in a future commit, but in the future if you can do that it would help.

I'll also add this info to the [style guide](http://wopi.readthedocs.io/en/latest/contributing/style_guide.html).

File: docs/native/src/wopi_requirements.rst (hunk 7 above)
Comment (Olga-MSFT) [inline docs/native/src/wopi_requirements.rst]: I'll go ahead and list this as a requirement on the WOPI requirements again, and remove the references for alternate support in the operational requirements page.
